lib/api/CAnomalyJobConfig.cc

```diff
@@ -634,9 +672,7 @@ void CAnomalyJobConfig::CAnalysisConfig::CDetectorConfig::parse(
 
     // The detector index is of type int for historical reasons
     // and for consistency across the code base.
-    // The explicit cast is required here as our JSON parser does not support
-    // 32 bit integer values.
-    m_DetectorIndex = static_cast<int>(parameters[DETECTOR_INDEX].as<std::ptrdiff_t>());
+    m_DetectorIndex = detectorIndex;
```

Comment:
I am not sure about this.  It means both the Java and C++ now rely on the assumption that the detectors array contains the detector indices in order.  It might create a problem in the future, for example if we decided to let users drag their detectors into a different order for display purposes but we wanted to remember their original indices so that we could still tie them up with the old results.  Somebody might change the logic for assigning detector indices in the Java code but not the C++, and then that would cause a subtle, hard-to-detect discrepancy between the detector index reported in C++ results and which detector it related to.

I can see why you've done it - it means you don't need to specify detector index in all the unit test config files.  And it's true that technically detector index doesn't have to be set for configs in the index that date back to 5.x and have never been rewritten.  However, these would get rewritten by the Java as they're written to the input file of the C++.

Would it be possible to take the parsed value if it exists and only use the array index if there is no parsed value?

Maybe something like:

```
    m_DetectorIndex = parameters[DETECTOR_INDEX].as<int>().fallback(fallbackDetectorIndex);
```

I imagine it wouldn't be hard to add support for `int` to the parser either to avoid the cast and comment.